tests/garage/envs/bullet/test_bullet_env.py

```diff
@@ -0,0 +1,90 @@
+"""Test module for BulletEnv"""
+import pickle
+
+import gym
+import pybullet_envs
+from pybullet_utils.bullet_client import BulletClient
+import pytest
+
+from garage.envs.bullet import BulletEnv
+from tests.helpers import step_env
+
+
+@pytest.mark.parametrize('env_ids', [pybullet_envs.getList()])
+def test_can_step(env_ids):
+    """Test Bullet environments can step"""
+
+    for env_id in env_ids:
+        # extract id string
+        env_id = env_id.replace('- ', '')
+        if env_id == 'KukaCamBulletEnv-v0':
+            # Kuka environments calls py_bullet.resetSimulation() in reset()
+            # unconditionally, which globally resets other simulations. So
+            # only one Kuka environment is tested.
+            continue
+        if env_id == 'RacecarZedBulletEnv-v0':
+            env = BulletEnv(gym.make(env_id, renders=False))
+        else:
+            env = BulletEnv(gym.make(env_id))
+        ob_space = env.observation_space
+        act_space = env.action_space
+        env.reset()
+
+        ob = ob_space.sample()
+        assert ob_space.contains(ob)
+        a = act_space.sample()
+        assert act_space.contains(a)
+        # Skip rendering because it causes TravisCI to run out of memory
+        step_env(env, render=False)
+        env.close()
+
+
+@pytest.mark.parametrize('env_ids', [pybullet_envs.getList()])
+def test_pickleable(env_ids):
+    """Test Bullet environments are pickleable"""
+    for env_id in env_ids:
+        # extract id string
+        env_id = env_id.replace('- ', '')
+        if env_id == 'RacecarZedBulletEnv-v0':
+            env = BulletEnv(gym.make(env_id, renders=False))
+        else:
+            env = BulletEnv(gym.make(env_id))
```

Comment:
Why not always set `renders=False`?

Reply:
By default all classes don't render the visualization. The only exception is class `RacecarZedBulletEnv`. 

However, some classes have the parameter name as `render` instead of `renders`. See [here](https://github.com/bulletphysics/bullet3/blob/master/examples/pybullet/gym/pybullet_envs/minitaur/envs/minitaur_gym_env.py#L92). Some classes don't have `render` for their constructors. See [here](https://github.com/bulletphysics/bullet3/blob/master/examples/pybullet/gym/pybullet_envs/gym_pendulum_envs.py#L10).

So the if/else logic here is necessary anyways. It's just easier to make an exception for `RacecarZedBulletEnv` instead of enumerating the other cases.

Reply:
I see. This inconsistency in pybullet is really painful.